dace/dtypes.py

```diff
@@ -403,9 +403,10 @@ def result_type_of(lhs, *rhs):
 
     rhs = rhs[0]
 
-    # Extract the type if symbolic
-    lhs = lhs.dtype if type(lhs).__name__ == 'symbol' else lhs
```

Comment:
Why do we check the name of the type, instead of something along the lines of `isinstance(lhs, (dace.symbol, sympy.symbol))`?

Reply:
Import loop

Reply:
Import loop